editor/scene_tree_dock.cpp

```diff
@@ -974,60 +972,69 @@ void SceneTreeDock::_tool_selected(int p_tool, bool p_confirm_override) {
 				break;
 			}
 
-			List<Node *> selection = editor_selection->get_selected_node_list();
+			EditorUndoRedoManager *undo_redo = EditorUndoRedoManager::get_singleton();
+			undo_redo->create_action(TTR("Replace with Branch Scenes"));
 
-			if (selection.size() != 1) {
-				accept->set_text(vformat(TTR("Saving the branch as a scene requires selecting only one node, but you have selected %d nodes."), selection.size()));
-				accept->popup_centered();
-				break;
-			}
+			List<Node *> selection = editor_selection->get_selected_node_list();
+			Ref<DirAccess> da = DirAccess::create(DirAccess::ACCESS_RESOURCES);
 
-			Node *tocopy = selection.front()->get();
+			for (Node *tocopy : selection) {
+				if (tocopy == scene) {
+					accept->set_text(TTR("Can't save the root node branch as an instantiated scene.\nTo create an editable copy of the current scene, duplicate it using the FileSystem dock context menu\nor create an inherited scene using Scene > New Inherited Scene... instead."));
+					accept->popup_centered();
+					continue;
+				}
 
-			if (tocopy == scene) {
-				accept->set_text(TTR("Can't save the root node branch as an instantiated scene.\nTo create an editable copy of the current scene, duplicate it using the FileSystem dock context menu\nor create an inherited scene using Scene > New Inherited Scene... instead."));
-				accept->popup_centered();
-				break;
-			}
+				if (tocopy != editor_data->get_edited_scene_root() && !tocopy->get_scene_file_path().is_empty()) {
+					accept->set_text(TTR("Can't save the branch of an already instantiated scene.\nTo create a variation of a scene, you can make an inherited scene based on the instantiated scene using Scene > New Inherited Scene... instead."));
+					accept->popup_centered();
+					continue;
```

Comment:
This means that in case of multiple errors, the dialog will repeatedly set text and popup. While in the end it works, I wonder if there is a better way to do it 🤔

Reply:
Do you have any suggestions in mind? My thought is to show a popup that says "The following nodes were not copied" with the node names, but that wouldn't give a specific reason why. Another idea is just appended each node and its error to a string and display potentially a large amount of text in the popup. That doesn't sound like the best solution though either. Open to suggestions

Reply:
Maybe some sort of bitmask/bool set for errors and if error appears, mark it in the set. Then after the loop you could check which error was flagged, add them to the dialog and display it all.
The affected scenes could be listed simply, but with a limit to e.g. 5 entries. So in the end the message would be:
```
These nodes failed to be saved: Node1, Node2, Node3, Node4, Node5 and X more nodes.
- Can't save the branch of an already instantiated scene.\nTo create a variation of a scene, you can make an inherited scene based on the instantiated scene using Scene > New Inherited Scene... instead.
- Etc.
```
Perfectly each failed node could be listed next to fail reason, but since the errors are so long, it's unfeasible for a simple dialog and improving it is out of scope now I'd say.